### DISCUSSION TITLE
How to make full use of custom tools and also a really simple problem I cant figure out

### DISCUSSION THREAD

**Discussion by brucewestfall** [zim-desktop-wiki/zim-desktop-wiki#2642] [Q&A]
I would love to see uses and techniques you folks have come up with for custom tools.  Please post what youve done and how you did it.  
Here is my simple question:
Previously I had a tool that sent a command to my terminal.  It was something like this.
`xfce4-terminal "python3 /home/bruce/CreateJob.py"`
I've tried single quotes, double, no quotes, asked chatgpt and i just get errors.  I know it will work because I did it before (and deleted it.Duh.)
I know I can make it into a script, but it just really bugs me this doesnt work

**Comment by maclomhair** [comment]
To see why it doesn't work, try running zim from command line with debug options enabled, ie. `zim --verbose`, or try looking at the Debug Log in "Tools->View Debug Log". AFAIU, zim will split command line arguments at spaces, but respect double quotes. I've had success with custom tools like such:
`/bin/bash -c "i=something; something else with \"escaped arguments\""`
